libglusterfs/src/iobuf.c

```diff
@@ -866,8 +832,7 @@ __iobref_grow(struct iobref *iobref)
 int
 __iobref_add(struct iobref *iobref, struct iobuf *iobuf)
 {
-    int i = 0;
-    int ret = -ENOMEM;
+    int ret = -EINVAL;
 
     GF_VALIDATE_OR_GOTO("iobuf", iobref, out);
```

Comment:
And now I think the validations are not needed, as they are done by the caller already.

Reply:
OK, this is called from the only place indeed.